### PR TITLE
SIRo Oracle Nav Fixes

### PR DESCRIPTION
**Pull request by ASzot**
Fixes for oracle navigation.

### CODE REVIEW

File: habitat-lab/habitat/config/default_structured_configs.py [hunk 1]
```diff
@@ -293,6 +293,7 @@ class OracleNavActionConfig(ActionConfig):
     ang_speed: float = 10.0
     allow_dyn_slide: bool = True
     allow_back: bool = True
+    # A value of -1.0 means we will get as close to the object as possible.
```
Comment (xavierpuigf) [inline habitat-lab/habitat/config/default_structured_configs.py:296]: How do we define as close to the object as possible? Maybe a short second sentence to clarify.

File: habitat-lab/habitat/tasks/rearrange/multi_task/pddl_predicate.py
Comment (xavierpuigf) [inline habitat-lab/habitat/tasks/rearrange/multi_task/pddl_predicate.py:1]: What do these changes do? Is there a reason to change this structure to dictionary?

File: habitat-lab/habitat/tasks/rearrange/utils.py [hunk 2]
```diff
@@ -441,6 +490,10 @@ def get_robot_spawns(
         if target_distance > distance_threshold or not is_navigable:
             continue
 
+        island_radius = sim.pathfinder.island_radius(start_position)
```
Comment (xavierpuigf) [inline habitat-lab/habitat/tasks/rearrange/utils.py:493]: Can you add a comment of what this does?

File: habitat-lab/habitat/tasks/rearrange/utils.py
Comment (vincentpierre) [inline habitat-lab/habitat/tasks/rearrange/utils.py:1]: Add return types

**Review by vincentpierre — COMMENTED** [review]
A lot of the comments I left [here](https://github.com/facebookresearch/habitat-lab/pull/1209) also apply.

File: habitat-lab/habitat/tasks/rearrange/utils.py (hunk 2 above)
Comment (jimmytyyang) [reply habitat-lab/habitat/tasks/rearrange/utils.py:493]: Same here! Thank you, Andrew!

Comment (jimmytyyang) [reply habitat-lab/habitat/tasks/rearrange/utils.py:493]: Same here! Thank you, Andrew!

File: habitat-baselines/habitat_baselines/config/rearrange/rl_hierarchical_oracle_nav.yaml [hunk 3]
```diff
@@ -13,6 +13,10 @@ defaults:
   - _self_
 
 habitat:
+  task:
+    actions:
+      oracle_nav_action:
```
Comment (jimmytyyang) [inline habitat-baselines/habitat_baselines/config/rearrange/rl_hierarchical_oracle_nav.yaml:18]: Hi Andrew! Could you please clarify what is the difference between oracle_nav and teleporting actions? Thank you!

**Review by jimmytyyang — COMMENTED** [review]
Looks good to me. i left some comments, thank you!

File: habitat-lab/habitat/tasks/rearrange/utils.py (hunk 2 above)
Comment (ASzot) [reply habitat-lab/habitat/tasks/rearrange/utils.py]: Removed this.

File: habitat-lab/habitat/tasks/rearrange/utils.py
Comment (ASzot) [reply habitat-lab/habitat/tasks/rearrange/utils.py:1]: Addressed in other PR.

File: habitat-baselines/habitat_baselines/config/rearrange/rl_hierarchical_oracle_nav.yaml (hunk 3 above)
Comment (ASzot) [reply habitat-baselines/habitat_baselines/config/rearrange/rl_hierarchical_oracle_nav.yaml:18]: Oracle nav means the navigation is simulated but takes the shortest path. Teleporting actions instantaneously moves the robot around.

File: habitat-lab/habitat/config/default_structured_configs.py (hunk 1 above)
Comment (ASzot) [reply habitat-lab/habitat/config/default_structured_configs.py:320]: This is addressed in this PR actually: https://github.com/facebookresearch/habitat-lab/pull/1209

File: habitat-lab/habitat/tasks/rearrange/utils.py [hunk 4]
```diff
@@ -388,6 +388,54 @@ def write_gfx_replay(gfx_keyframe_str, task_config, ep_id):
         text_file.write(gfx_keyframe_str)
 
 
+def place_agent_at_dist(
```
Comment (akshararai) [inline habitat-lab/habitat/tasks/rearrange/utils.py:391]: why is this `place_agent_at_dist` and not `place_agenta_at_pos`?

**Review by akshararai — APPROVED** [review]
looks good overall, thanks for fixing these things. I left some minor comments, but looks good to me.

File: habitat-lab/habitat/tasks/rearrange/utils.py (hunk 4 above)
Comment (ASzot) [reply habitat-lab/habitat/tasks/rearrange/utils.py:391]: Sure, I will rename to `place_agent_at_dist_from_pos`.

File: habitat-baselines/habitat_baselines/config/experiments_hab3/single_agent_kinematic_oracle.yaml [hunk 5]
```diff
@@ -0,0 +1,38 @@
+# @package _global_
+
+defaults:
+  - /rearrange/rl_hierarchical_oracle_nav
+  - override /habitat_baselines/rl/policy: hab3_hl_neural
+  - override /habitat_baselines/rl/policy/hierarchical_policy/defined_skills: oracle_skills
```
[the remark this answers is not in the document]
Comment (ASzot) [reply habitat-baselines/habitat_baselines/config/experiments_hab3/single_agent_kinematic_oracle.yaml:6]: There is not much difference. The base override is different and there is an override for the oracle nav target distance.

**Pull request merged**